File: Serene3/Serene3.Web/Modules/com/states/statesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Serene3.com.Columns;

[ColumnsScript("com.states")]
[BasedOnRow(typeof(statesRow), CheckNames = true)]
public class statesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string CountryName { get; set; }
}

File: Serene3/Serene3.Web/Modules/ServerTypes/com/statesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { statesRow } from "./statesRow";

export interface statesColumns {
    Id: Column<statesRow>;
    Name: Column<statesRow>;
    CountryName: Column<statesRow>;
}

export class statesColumns extends ColumnsBase<statesRow> {
    static readonly columnsKey = 'com.states';
    static readonly Fields = fieldsProxy<statesColumns>();
}